docs/api.md

```diff
@@ -589,3 +589,241 @@ state.
+field. The values must be within the limits as defined in the *FirmwareSchema*.
+Only settings which are not defined as `ReadOnly` or `Unique` (such as
+SerialNumbers) will be accepted.
+
+### HostFirmwareSettings status
+
+The *HostFirmwareSettings's* *status* defines the actual BIOS settings read
+from Ironic.
+
+#### schema
+
+`Schema` is a reference to a *FirmwareSchema* resource that describes each
+BIOS setting by type and configurable limits for the type.
+
+#### settings
+
+The `settings` are an array of name/value pairs listing the complete set
+of BIOS settings retrieved from Ironic.
+
+#### conditions
+
+`conditions` reflects the status of the fields in the *spec*. Possible
+conditions are:
+
+* *Valid* -- When set to *True* indicates that all *settings* in the *spec* are
+  correct for names and values. This is also set to *True* when nothing is set
+  in the *spec*. When set to *False* indicates that one or more names or values
+  in the *spec* are incorrect, the actual error will be shown in the *Events*
+  field.
+* *UpdateRequested* -- Indicates whether or not settings in the *spec* are
+  different than settings in the *status*. When set to *True* the settings that
+  are different will be included in the clean-steps that are written to Ironic
+  as part of cleaning.
+
+### HostFirmwareSettings Example
+
+The following is a complete example from a running cluster of a
+*HostFirmwareSettings* resource (in YAML), it includes its spec and status
+sections:
+
+apiVersion: metal3.io/v1alpha1
+kind: HostFirmwareSettings
+metadata:
+  creationTimestamp: "2021-11-03T21:21:02Z"
+  generation: 1
+  name: ostest-worker-0
+  namespace: openshift-machine-api
+  ownerReferences:
+  - apiVersion: metal3.io/v1alpha1
+    blockOwnerDeletion: true
+    controller: true
+    kind: BareMetalHost
+    name: ostest-worker-0
+    uid: 87b08d19-b94c-4a29-901a-67890d9eb843
+  resourceVersion: "24688"
+  uid: 293a351b-1743-437d-abed-da53d41d8804
+spec:
+  settings: {}
+status:
+  conditions:
+  - lastTransitionTime: "2021-11-03T21:21:54Z"
+    message: ""
+    observedGeneration: 1
+    reason: Success
+    status: "False"
+    type: UpdateRequested
+  - lastTransitionTime: "2021-11-03T21:21:54Z"
+    message: ""
+    observedGeneration: 1
+    reason: Success
+    status: "True"
+    type: Valid
+  schema:
+    name: schema-dc98d7c8
+    namespace: openshift-machine-api
+  settings:
+    BootMode: Uefi
+    EmbeddedSata: Raid
+    L2Cache: 10x256 KB
+    NicBoot1: NetworkBoot
+    NumCores: "10"
+    ProcTurboMode: Enabled
+    QuietBoot: "true"
+    SecureBootStatus: Enabled
+    SerialNumber: QPX12345
+
+## FirmwareSchema
+
+A **FirmwareSchema** resource contains the limits each setting, specific to
+each host.  This data comes directly from the BMC via Ironic. It can be used
+to prevent misconfiguration of the **HostFirmwareSettings** *spec* field so
+that can invalid values are not sent to the host.
```

Comment:
Maybe it could be useful to specify the granularity of the schema resource (ie before it was specified for the hfs: https://github.com/metal3-io/baremetal-operator/pull/1015/files#diff-9eddf4dc51bf6a0125ca7fb094468ad284112270385c41cebce4f8f0a29620abR596)

Reply:
Agree, fixed.